maven-core/src/main/java/org/apache/maven/DefaultMaven.java

```diff
@@ -232,25 +232,40 @@ private MavenExecutionResult doExecute( MavenExecutionRequest request, MavenSess
             return addExceptionToResult( result, e );
         }
 
-        WorkspaceReader reactorWorkspace;
+        List<WorkspaceReader> workspaceReaders = new ArrayList<>();
+        WorkspaceReader repoWorkspaceReader = repoSession.getWorkspaceReader();
+        //
+        // Desired order of precedence for local artifact repositories
+        //
         try
         {
-            reactorWorkspace = container.lookup( WorkspaceReader.class, ReactorReader.HINT );
+            // 1) Reactor workspace reader
+            workspaceReaders.add( container.lookup( WorkspaceReader.class, ReactorReader.HINT ) );
         }
         catch ( ComponentLookupException e )
         {
             return addExceptionToResult( result, e );
         }
-
-        //
-        // Desired order of precedence for local artifact repositories
-        //
-        // Reactor
-        // Workspace
+        if ( repoWorkspaceReader != null )
+        {
+            // 2) Repository system session scoped workspace reader
+            workspaceReaders.add( repoWorkspaceReader );
+        }
+        for ( WorkspaceReader workspaceReader : getProjectScopedExtensions( session.getProjects(),
+                                                                            WorkspaceReader.class ) )
+        {
+            if ( workspaceReaders.contains( workspaceReader ) )
+            {
+                continue;
+            }
+            // 3) .. n) Project scoped workspace reader
```

Comment:
This comment is misplaced compared to the master PR.

Reply:
Done